### DISCUSSION TITLE
thu-info-lib的网页插件还可以使用吗？似乎限于CORS无法成功请求

### DISCUSSION THREAD

**Discussion by SunnyCloudYang** [thu-info-community/thu-info-app#667] [Q&A] [answered]
插件使用`const helper = InfoHelper.InfoHelper(); helper.login({"userId": "xxxxxxxxxx", "password": "xxxxxxxx"});`时报错如下：
![image](https://github.com/user-attachments/assets/86eca5ef-4477-4d2e-b98e-b06ff397f24f)
另外是否有办法在React（web）项目中使用thu-info-lib？

**Comment by UNIDY2002** [comment, marked as answer]
一直没测插件，疏于维护了，需要在插件的 manifest.json 里面把 oauth.tsinghua.edu.cn 和 id.tsinghua.edu.cn 加进去。

另外，虽然从代码层面，lib 可以在 React 项目中引用，但跨域请求问题不可避免。可以考虑浏览器插件或微信小程序的形式。

**Reply by SunnyCloudYang** [reply]
插件确实只能在登录页面使用，或者在除浏览器之外的app或脚本等可以设置unsafe header的环境中使用，其他页面会被CORS限制无法避开。React+lib实测需要node>=20，且需要设置`--unsafeLegacyRenegotiation`。

最后helper真的很好用，感谢您的解答和杰出贡献🥳！
